enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,180 @@
+---
+title: neat-enhancement-idea
+authors:
+  - "@camilamacedo86"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-02-19
+last-updated: 2021-02-19
+status: implementable
+see-also:
+  - "/enhancements/olm/max-openshift-versions-for-operators.md"
+---
+
+# OpenshiftCatalogueValidator 
+
+This proposal describes a new set/suite of checks to be implemented in the [operator-framework/api][oper-api] to validate the bundle operator data regarding specific rules for the OpenShift Catalogue.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Currently, [operator-framework/api][oper-api] centralized the code implementation for superset/suites of checks used in the tools and across the board. See [here](https://github.com/operator-framework/api/blob/master/pkg/validation/validation.go#L38-L46) such as:
+
+- `BundleValidator` check if the operator bundle spec is accordingly to spec definition. For further information see [here](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md).
+- `OperatorHubValidator` checks of the operator is respecting the required criteria to be published in the [OperatorHub.io](https://operatorhub.io/).
+
+The new `OpenShiftCatalogueValidator` proposed here has the purpose of perform checks which are specifically to publish the operator bundle on the OpenShift Catalogue. The validator will raise issues or warnings accordingly.
+
+This checks will also be available to be used by the users via the tools such as SDK and consequently also called in the pipeline. Note that SDK provides currently:
+
+```sh
+To validate a bundle with a superset of requirements for operatorhub.io specifically:
+   
+  $ operator-sdk bundle validate ./bundle --select-optional name=operatorhub
+``` 
+
+Then, the idea proposed here would allow the tool also provide:
+
+```sh
+To validate a bundle with a superset of requirements for OpenShift Catalogue specifically:
+   
+$ operator-sdk bundle validate ./bundle --select-optional name=openshiftcatalogue
+``` 
+
+**NOTE** This validator and its usage are demanding to address a further requirement proposal to audit all bundles of a provided catalogue.
+
+## Motivation
+
+- Provide an operator bundle validator to raise issues and warnings regarding specific configurations and specifications which are only valid for `OpenShift/OKD` cluster types
```

Comment:
My idea here is that the OpenshiftValidator will do any check to ensure any criteria which are specific for OCP vendor. Then, it can be related to the OCP versions supportability as any other scenario that we might have in the future. E.g a specific spec/annotation that is only required when the operator will be published in OCP.

However, if the bundle is configured to work only in OCP 4.5 we should consider that to make the checks as well. I will add your suggestion and try to clarify it better.

Reply:
I changed for `Provide an operator bundle validator to raise issues and warnings regarding specific configurations and specifications which are only valid for `OpenShift/OKD` cluster types accordingly with the OCP versions supported by the bundle.`

Reply:
addressed.